File: nom-ui/src/app/core/models/meal-plan-entry.model.ts
export interface MealPlanEntry {
  id: number;
  recipeId: number | null;
  recipeName: string | null;
  recipeImage: string | null;
  /** Set when this slot holds a standalone whole food (apple, protein bar) instead of a recipe. */
  ingredientId?: number | null;
  ingredientName?: string | null;
  quantity?: number | null;
  measurementId?: number | null;
  measurementName?: string | null;
  foodGroupId?: number | null;
  foodGroupName?: string | null;
  title: string | null;
  notes: string | null;
  calories: number | null;
  proteinGrams: number | null;
  carbGrams: number | null;
  fatGrams: number | null;
  completedDate: string | null;
  shoppingCompletedAt: string | null;
}

File: nom-api/Nom.Orch/Models/MealPlan/MealPlanEntryModel.cs
using System;

namespace Nom.Orch.Models.MealPlan
{
    public class MealPlanEntryModel
    {
        public long Id { get; set; }
        public long? RecipeId { get; set; }
        public string? RecipeName { get; set; }
        public string? RecipeImage { get; set; }

        /// <summary>Set when this slot holds a standalone whole food (apple, protein bar) instead of a recipe.</summary>
        public long? IngredientId { get; set; }
        public string? IngredientName { get; set; }
        public decimal? Quantity { get; set; }
        public long? MeasurementId { get; set; }
        public string? MeasurementName { get; set; }
        public long? FoodGroupId { get; set; }
        public string? FoodGroupName { get; set; }

        public string? Title { get; set; }
        public string? Notes { get; set; }
        public decimal? Calories { get; set; }
        public decimal? ProteinGrams { get; set; }
        public decimal? CarbGrams { get; set; }
        public decimal? FatGrams { get; set; }
        public DateOnly? CompletedDate { get; set; }
        public DateTime? ShoppingCompletedAt { get; set; }
    }
}
